### DISCUSSION TITLE
Is there public list with plans for nearest future?

### DISCUSSION THREAD

**Discussion by vchigrin** [InfiniTimeOrg/InfiniTime#2199] [General]
Just noticed comment in one of PRs (https://github.com/InfiniTimeOrg/InfiniTime/pull/375#issuecomment-1100837137) and decided reply on it here since discussion did not relate to that PR. As comment author noted, not all PR-s can be merged due technical limitations of hardware and because one features considered more important then others.

Is there place to look for "what work considered as priority number one"? It may be useful for developers, that have some free time and energy, so their work will bring most benefit to the project. If such list exist, it will be cool to distinguish assigned vs not yet assigned tasks, to decrease likelihood of conflicts. I saw the "milestones" list, but looks like it contains already made,  mostly finished PRs.

I understand that this is non-commercial project that people develop in their free time. I just think that everybody will be more happy if potential contributes will make work that maintainers consider most useful.

**Comment by mark9064** [comment]
I'm working to try and burn through all the PRs that are currently open, it will take time to get there though

Most features and improvements suggested can be integrated into InfiniTime. Of course there are some which are too large or  impossible due to hardware but these are usually not the case - most things can be iterated on until they align nicely with the InfiniTime vision with good quality code. With the new optional app/watchface system large apps can still be merged but be off by default.

Review time and speed is the biggest reason why there are so many PRs just sitting around. For anyone who knows C++ and/or the codebase, doing review on other people's PRs is the absolute best way to shift load off of maintainers and increase development velocity. There's no need to be an expert to do review - even small things like pointing out typos help massively. Discovering how hard review can be comes quite quickly after getting started! I promise the maintainers are not just super lazy :P

The milestones are used to track PRs which will be merged before the next release (provided there aren't unexpected problems). Most PRs queued onto a milestone already have a clear path to being merged as you say. These PRs very much still need review though!

As for features to be made, the discussions board contains the most requested ones. Most of the ones that don't already have a PR drafted are quite hard though. For example, [sorting by top](https://github.com/InfiniTimeOrg/InfiniTime/discussions?discussions_q=is%3Aopen+sort%3Atop) gives
- sleep tracking (huge feature, in progress with SleepTk port)
- remove messages on remote device (requires new bluetooth protocol that there is no standard for, plus reworking of both InfiniTime and companion apps to implement)
- screen rotation (requires hard changes with the LCD panel, things like scrolling and animations currently make assumptions about the orientation iirc)
- longer timer support (this one looks quite doable actually, but probably requires some UI rework to keep the buttons usable)
- smart home app (again requires new BLE protocol, app, and companion app integration)

A lot of these are difficult features. There's also the open PRs (and some issues predating the discussion forum) which you can loosely prioritise by thumbsup/heart counts.

There are also lots of important but behind-the-scenes things to work on. For example I would like to migrate off the NRF SDK to nrfx, use upstream FreeRTOS, move to latest LVGL 9, and update NimBLE. But all of these are really time consuming, and some of them such as LVGL may not even be feasible (due to increasing memory demands in this case)!

---

So what to do? To anyone who wants to help with code and knows the codebase: please help with review! Once we work through the backlog of open PRs, things will be much more manageable. Aside from that, I would look at the most popular feature requests and get involved: start discussing how they can turn into real apps and services inside InfiniTime. Once a discussion is going, it's much easier to review someone else's proposal than come up with one from scratch IMO

Maybe there could be some central list of features? But this is really the intention of the discussions page, we should rework it if it's not fulfilling its goal
